animation_nodes/data_structures/meshes/mesh_data.pyx

```diff
@@ -370,3 +384,272 @@ def calculateLinkedVertices(int verticesAmount, EdgeIndicesList edges):
+
+    cdef float alpha = dotVec3(&uw, &n) / normSq
+    cdef float beta = dotVec3(&wv, &n) / normSq
+    cdef float gamma = 1 - alpha - beta
+    if alpha >= 0 and alpha <= 1 and beta >= 0 and beta <= 1 and gamma >= 0 and gamma <= 1:
+        return True
+    else:
+        return False
+
+# Transformation of vertices of polygon into xy-plane
+@cython.cdivision(True)
+cdef transformVerticesOfPolygon(Vector3DList vertices, LongList polygonIndices):
+    cdef Py_ssize_t polyLength = polygonIndices.length
+    cdef Vector3DList polyVertices = Vector3DList(length = polyLength)
+
+    # Compute polygon normal with Nowell's method
+    cdef Vector3 polyNormal = toVector3((0, 0, 0))
+    cdef Py_ssize_t i
+    for i in range(polyLength):
+        v1 = vertices.data[polygonIndices.data[i]]
+        v2 = vertices.data[polygonIndices.data[(i + 1) % polyLength]]
+
+        polyNormal.x += (v1.y - v2.y) * (v1.z + v2.z)
+        polyNormal.y += (v1.z - v2.z) * (v1.x + v2.x)
+        polyNormal.z += (v1.x - v2.x) * (v1.y + v2.y)
+
+        polyVertices.data[i] = v1
+    normalizeVec3_InPlace(&polyNormal)
+
+    # Calculate the coefficient for Rodrigues's rotation formula
+    cdef Vector3 zAxis = toVector3((0, 0, 1))
+    cdef Vector3 rotAxis
+    crossVec3(&rotAxis, &polyNormal, &zAxis)
+    normalizeVec3_InPlace(&rotAxis)
+
+    cdef float angle = angleNormalizedVec3(&polyNormal, &zAxis)
+    cdef float sint = sin(angle)
+    cdef float cost = cos(angle)
+    cdef float fact = 1 - cost
+
+    # Rotating vertices of polygon, and droping the z-comp
+    cdef Vector3DList rotPolyVertices = Vector3DList(length = polyLength)
+    cdef Vector3 vertex, cross, rotPolyVertex
+    cdef float dot
+    for i in range(polyLength):
+        vertex = polyVertices.data[i]
+        crossVec3(&cross, &rotAxis, &vertex)
+        dot = dotVec3(&rotAxis, &vertex)
+
+        rotPolyVertex.x = cost * vertex.x + sint * cross.x + dot * fact * rotAxis.x
+        rotPolyVertex.y = cost * vertex.y + sint * cross.y + dot * fact * rotAxis.y
+        rotPolyVertex.z = cost * vertex.z + sint * cross.z + dot * fact * rotAxis.z
+        rotPolyVertices.data[i].x = rotPolyVertex.x
+        rotPolyVertices.data[i].y = rotPolyVertex.y
+        rotPolyVertices.data[i].z = 0
+
+    # Compute avg. vector
+    cdef Vector3 avgVector = toVector3((0, 0, 0))
+    for i in range(polyLength):
+        avgVector.x += rotPolyVertices.data[i].x
+        avgVector.y += rotPolyVertices.data[i].y
+        avgVector.z += rotPolyVertices.data[i].z
+    scaleVec3_Inplace(&avgVector, 1.0 / polyLength)
+
+    # Bring vertices of polygon to the world center
+    cdef Vector3DList offsetPolyVertices = Vector3DList(length = polyLength)
+    for i in range(polyLength):
+        offsetPolyVertices.data[i].x =  rotPolyVertices.data[i].x - avgVector.x
+        offsetPolyVertices.data[i].y =  rotPolyVertices.data[i].y - avgVector.y
+        offsetPolyVertices.data[i].z =  rotPolyVertices.data[i].z - avgVector.z
+
```

Comment:
Is this really necessary? Can't we just skip it?

Reply:
I have done testing, there is no longer need of this offset. I'll remove it.

Reply:
Done.